Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -1,13 +1,14 @@
 import Foundation
+import Combine
 
 /// A global singleton which holds base configuration for the OpenFeature library.
 /// Configuration here will be shared across all ``Client``s.
-public class OpenFeatureAPI {
+public class OpenFeatureAPI: GlobalEventPublisher {
     private var _provider: FeatureProvider?
     private var _context: EvaluationContext?
     private(set) var hooks: [any Hook] = []
-
-    private let providerNotificationCentre = NotificationCenter()
+    private var providerObserver: AnyCancellable?
+    private var globalEventState = PassthroughSubject<ProviderEvent, Never>()
```

Comment:
`PassthroughSubject` (doesn't store the latest observed event) instead of `CurrentValueSubject` (stores the latest observed event), because in a global context (and assuming multiple providers set) it's not that useful to get the latest event observed by any arbitrary provider